### PR TITLE
Missing asterisk operator for unpacking extra values

### PR DESCRIPTION
**Pull request by NaleRaphael**
See also #24.

### CODE REVIEW

File: torch_lr_finder/lr_finder.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def _validate(self, dataloader):
         running_loss = 0
         self.model.eval()
         with torch.no_grad():
-            for inputs, labels in dataloader:
+            for inputs, labels, *_ in dataloader:
```
Comment (davidtvs) [inline torch_lr_finder/lr_finder.py:289]: This is not needed since the `__next__` method only returns the pair `(inputs, labels)`

Comment (NaleRaphael) [reply torch_lr_finder/lr_finder.py:289]: But would it be safer to keep it in case someone use a dataloader with custom `DataLoaderIter`?

Like the implementation shown in the link below, `self._next_data()` will be invoked in `__next__`, but the actual content of `data` returned by `self._next_data()` is determined by user's implementation. (I have done something similar before 😆)
https://github.com/pytorch/pytorch/blob/7f73f1d/torch/utils/data/dataloader.py#L344

Comment (davidtvs) [reply torch_lr_finder/lr_finder.py:289]: Ya, you are right. I was thinking that `dataloader` was wrapped in `DataLoaderIterWrapper` but it isn't; so it's perfectly possible for it to return additional values.

**Comment by NaleRaphael** [comment]
Oh, BTW, if you need some help for writing unit test, I will be glad to do it.
Maybe I could finish those parts written by me at first.
And if you need to develop a basic architecture of test at the beginning, just feel free to let me know when it is done!

**Comment by NaleRaphael** [comment]
Here is a script to run LRFinder with/without `val_loader`, and you can change the variable `ex_size` to control the number of extra values returned by `SimpleData` iterator. Hope this helps!


```python
import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
from torch.utils.data import Dataset, Subset, DataLoader
from torch_lr_finder import LRFinder


class SimpleNet(nn.Module):
    def __init__(self):
        super(SimpleNet, self).__init__()
        self.net = nn.Sequential(
            nn.Linear(8, 4),
            nn.Linear(4, 1),
        )

    def forward(self, x):
        return self.net(x).squeeze()


class SimpleData(Dataset):
    def __init__(self, size, ex_size=0):
        self.data = torch.rand((size, 10, 8))
        self.label = torch.randint(1, 10, (size,))
        if ex_size:
            self.ex_info = [torch.randint(0, 1, (size,)) for i in range(ex_size)]
        else:
            self.ex_info = None

    def __getitem__(self, index):
        if self.ex_info is not None:
            retval = [self.data[index], self.label[index]]
            retval.extend([v[index] for v in self.ex_info])
            return retval
        else:
            return self.data[index], self.label[index]

    def __len__(self):
        return len(self.data)


if __name__ == "__main__":
    ex_size = 2     # for dummy extra data
    dataset = SimpleData(256, ex_size=ex_size)
    trainset = Subset(dataset, range(224))
    validset = Subset(dataset, range(224, 256))

    batch_size = 8
    trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True)
    valloader = DataLoader(validset, batch_size=batch_size)

    print('--- shape of sample batch ---')
    sample = next(iter(trainloader))
    print('data: {}, label: {}'.format(sample[0].shape, sample[1].shape))
    if ex_size:
        print('extra data: {}'.format([v.shape for v in sample[2:]]))
    else:
        print('no extra data.')
    print('-----------------------------')

    model = SimpleNet()

    device = torch.device('cuda')
    model = model.to(device)
    optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.5)
    criterion = nn.NLLLoss()

    lr_finder = LRFinder(model, optimizer, criterion, device='cuda')
    lr_finder.range_test(
        trainloader, end_lr=1, num_iter=100, step_mode='exp',
        start_lr=1e-5, val_loader=valloader
    )
    lr_finder.plot()
    lr_finder.reset()
```

**Comment by davidtvs** [comment]
Not really familiar with the Python unit testing frameworks but from what I know, creating a test folder in the root of the repository and implementing tests using the unittest package seems like a good first step.

Thanks for offering to help really appreciate it 👍 . Please feel free to implement some tests - something like that script would be a good start I guess. Let's do it a different PR though.

This PR looks good to me. I'll merge and close it.
